template.yaml

```diff
@@ -72,32 +126,102 @@ Resources:
         Variables:
           BACKFILL_DAYS:
             Ref: BackfillDays
+          DBNAME:
+            Ref: DatabaseName
+          ENABLE_INFREQUENT_CHANNELS:
+            Ref: EnableInfrequentChannels
+          DECRYPT_KEYS:
+            Fn::If:
+              - create_secret
+              - 'true'
+              - 'false'
           ENABLE_CONSOLE_LOG: 'false'
           ENVIRONMENT: aws
           HTTP_LOGS_ENDPOINT:
-            Ref: HttpLogsEndpoint
+            Fn::If:
+              - create_secret
+              - Ref: SumoLogicSecret
+              - Ref: HttpLogsEndpoint
           TOKEN:
-            Ref: Token
-          ENABLE_INFREQUENT_CHANNELS:
-            Ref: EnableInfrequentChannels
-          INFREQUENT_CHANNELS_THRESHOLD_IN_HOURS:
-            Ref: InfrequentChannelsThresholdInHours
-          INFREQUENT_CHANNELS_MESSAGES_FETCH_TIME_IN_HOURS:
-            Ref: InfrequentChannelsMessagesFetchTimeInHours
-          DBNAME:
-            Ref: DatabaseName
+            Fn::If:
+              - create_secret
+              - Ref: SumoLogicSecret
+              - Ref: Token
       Events:
         ScheduleRule:
           Properties:
             Schedule: rate(10 minutes)
           Type: Schedule
       Handler: main.main
-      Layers:
-      - Fn::Sub: arn:aws:lambda:${AWS::Region}:956882708938:layer:Slack-deps:5
       MemorySize: 256
       Policies:
-      - AmazonDynamoDBFullAccess
+        - AmazonDynamoDBFullAccess
```

Comment:
This policy is already defined in the Role so I guess we can remove it.

Reply:
The role policy has to be removed, as the policy attached is used.

Reply:
but I think we should be good as the change would be in sdk also